services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -416,7 +416,7 @@ async def test_outputs_watcher_disabling(
     outputs_context: OutputsContext = test_client.application.state.outputs_context
     outputs_manager: OutputsManager = test_client.application.state.outputs_manager
     outputs_manager.task_monitor_interval_s = WAIT_FOR_OUTPUTS_WATCHER / 10
-    WAIT_FOR_EVENTS = outputs_manager.task_monitor_interval_s * 10
+    WAIT_FOR_EVENTS = outputs_manager.task_monitor_interval_s * 100
```

Comment:
Since some time we are trying to avoid sleeps as ``await asyncio.sleep(WAIT_FOR_EVENTS)`` and if there is no alternative (e.g. a signal-slot) we use instead short retries with a timeout. This way our tests do not increase unnecessarily in time!

Reply:
Look in the code for ``_TENACITY_RETRY_PARAMS``  to see a couple of examples

Reply:
OK, I've managed to make it work with tenacity. Looks way better now!